cli/src/package.rs

```diff
@@ -0,0 +1,156 @@
+//! CLI commands for package management.
+
+use std::{
+    collections::HashMap,
+    env::current_dir,
+    path::{Path, PathBuf},
+};
+
+use nickel_lang_core::{identifier::Ident, package::PackageMap};
+use nickel_lang_package::{config::Config, ManifestFile};
+
+use crate::{
+    error::{CliResult, Error},
+    global::GlobalContext,
+};
+
+#[derive(clap::Subcommand, Debug)]
+pub enum Command {
+    /// Generate a lock file for a package.
+    ///
+    /// The lock file contains exact information about a package's dependencies,
+    /// including recursive dependencies.
+    Lock {
+        /// The path at which to write the lock file.
+        ///
+        /// Defaults to the filename `package.ncl.lock` in the same directory
```

Comment:
It's a detail, but I think usually package managers don't keep the `.ncl` equivalent in the lock file: `flake.nix` -> `flake.lock`, `Cargo.toml` -> `Cargo.lock`, `Pipefile` -> `Pipefile.lock`, `foo_bar.gemspec` -> `Gemfile.lock`, etc.

Reply:
Right, but I was worried about colliding with npm's (the other npm :wink:) `package.lock`. Maybe it's better to avoid a collision by having a more nickel-specific name than "package", though...

Reply:
I see. Yes, maybe having a more specific name than "package" could be helpful in general, if only there are projects with both an NPM manifest an a Nickel manifest (or even imagine that you handle everything with Nickel, so you would like to write your "package.json" as a "package.ncl" instead and generate it).

Reply:
Ok, I realized while fiddling with the website that actually npm uses "package-lock.json", not "package.lock". So there would be no conflict to go with "package.lock", but maybe a more distinctive name is better anyway. What do you think about `Nickel.ncl`/`Nickel.lock`? Or `Nickelfile.ncl`/`Nickelfile.lock`?